packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -4505,6 +4505,21 @@ void main() {
       final Text hintTextWidget = tester.widget(hintTextFinder);
       expect(hintTextWidget.style!.overflow, decoration.hintStyle!.overflow);
     });
+
+    testWidgets('hint should be ignore its height when input text', (WidgetTester tester) async {
+      final String hintText = 'hint text' * 20;
+      final InputDecoration decoration = InputDecoration(
+        hintText: hintText,
+        maintainHintHeight: false,
+      );
+
+      await tester.pumpWidget(
+        buildInputDecorator(
+          decoration: decoration,
+        ),
+      );
+      expect(find.text(hintText), findsNothing);
```

Comment:
Rather than finding no hint text, I think it's better to test the widget height. Something like
```
    expect(tester.getSize(find.byType(InputDecorator)).height, lessThan(100));
```